classes/hypeJunction/Feed/Router.php

```diff
@@ -15,6 +15,12 @@ public static function handleActivityPage($segments) {
 
 		$page = array_shift($segments);
 
+		if (!$page) {
+			$tabs = Config::getTabs(true);
+			$main_tab = array_pop(array_reverse($tabs));
```

Comment:
= `array_shift`

Reply:
Yeah!
I read that...(but not consider the array_reverse() cost 😅)

the complexity of array_pop() is O(1). 
the complexity of array_shift() is O(n).

Maybe better reset()

Reply:
Got a link?

Reply:
Micro-optimization that affects readability 👎

Reply:
I don't think it even optimizes anything. I would think shifting first element of an array is quicker than reading the length and finding the last element.